File: codes/yujo/463_islandperimeter.ts
function islandPerimeter(grid: number[][]): number {
  let count = 0;
  const gridLength = grid.length;
  const gridItemLength = grid[0].length;
  for (let i = 0; i < gridLength; i++) {
    for (let j = 0; j < gridItemLength; j++) {
      if (grid[i][j] === 1) {
        count += 4;
        if (i > 0 && grid[i - 1][j] === 1) count -= 1;
        if (j > 0 && grid[i][j - 1] === 1) count -= 1;
        if (i < gridLength - 1 && grid[i + 1][j] === 1) count -= 1;
        if (j < gridItemLength - 1 && grid[i][j + 1] === 1) count -= 1;
      }
    }
  }
  return count;
}

File: codes/vimkim/463_islandperimeter.py
class Solution:
    def islandPerimeter(self, grid: List[List[int]]) -> int:
        N = len(grid)
        M = len(grid[0])

        count = 0
        directions = ((1, 0), (0, 1))

        for i in range(N):
            for j in range(M):
                if grid[i][j] == 1:
                    count += 4  # Add 4 sides for each land cell

                    for di, dj in directions:
                        ni, nj = i + di, j + dj

                        if 0 <= ni < N and 0 <= nj < M and grid[ni][nj] == 1:
                            count -= 2  # Subtract 2 sides for each adjacent land cell
        return count